frontend/www/js/omegaup/components/contest/Intro.vue

```diff
@@ -0,0 +1,311 @@
+  }
+
+  get differentStartsDescription(): string {
+    return UI.formatString(T.contestIntroDifferentStarts, {
+      window_length: this.formatTimeInRules(this.contest.window_length),
+    });
+  }
+
+  get scoreboardDescription(): string {
+    const contest = this.contest;
+    if (!contest.scoreboard || !contest.finish_time || !contest.start_time) {
+      return '';
+    }
+    if (contest.scoreboard === 100) {
+      return T.contestIntroScoreboardTimePercentOneHundred;
+    }
+    if (contest.scoreboard === 0) {
+      return T.contestIntroScoreboardTimePercentZero;
+    }
+    const minutesPercentage = Math.floor(
+      (contest.scoreboard / 100) *
+        ((contest.finish_time.getTime() - contest.start_time.getTime()) /
+          60000),
+    );
+    return UI.formatString(T.contestIntroScoreboardTimePercent, {
+      window_length: this.formatTimeInRules(minutesPercentage),
+    });
+  }
+
+  get submissionsGapDescription(): string {
+    if (!this.contest.submissions_gap) {
+      return '';
+    }
+    return UI.formatString(T.contestIntroSubmissionsSeparationDesc, {
+      window_length: this.formatTimeInRules(this.contest.submissions_gap / 60),
+    });
+  }
+
+  get penaltyDescription(): string {
+    return UI.formatString(T.contestIntroPenaltyDesc, {
+      window_length: this.formatTimeInRules(this.contest.penalty),
+    });
+  }
+
+  get pointsDecayDescription(): string {
+    return UI.formatString(T.contestNewFormDecrementFactor, {
+      window_length: this.contest.points_decay_factor,
+    });
+  }
+
+  get isButtonDisabled(): boolean {
+    return (
+      this.needsBasicInformation ||
+      (this.shareUserInformation === null &&
+        this.requestsUserInformation !== 'no') ||
+      (this.requestsUserInformation === 'required' &&
+        !this.shareUserInformation)
+    );
+  }
+
+  @Watch('timeLeft')
+  onValueChanged(newValue: number): void {
+    if (newValue <= 0) {
+      if (!this.timerInterval) return;
+      clearInterval(this.timerInterval);
+      this.timerInterval = 0;
+      this.now = Date.now();
+    }
+  }
+
+  startTimer(): void {
+    this.timerInterval = window.setInterval(() => (this.timePassed += 1), 1000);
+  }
+
+  formatTimeInRules(timeInMinutes?: number): string {
+    if (!timeInMinutes) {
+      return '';
+    }
+    const hours = Math.floor(timeInMinutes / 60);
+    if (hours <= 0) {
+      return timeInMinutes + 'm';
+    } else {
+      const minutes = timeInMinutes % 60;
+      return `${hours}h${minutes}m`;
+    }
+  }
+
+  onStartContest() {
+    const request = {
+      contest_alias: this.contest.alias,
+      share_user_information: this.shareUserInformation,
+    };
+    let userInformationRequest = {};
+    if (this.requestsUserInformation === 'required') {
+      userInformationRequest = {
+        privacy_git_object_id: this.statement.gitObjectId,
+        statement_type: this.statement.statementType,
+      };
+    }
+    $.extend(request, userInformationRequest);
```

Comment:
no se puede usar [`Object.assign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) para evitar a jQuery? ahí en ese link tienen un polyfill.

alternativamente,

```suggestion
    if (this.requestsUserInformation === 'required') {
      request.privacy_git_object_id = this.statement.gitObjectId;
      request.statement_type = this.statement.statementType;
    }
```

Reply:
Listo, usé la alternativa simple.